docs/components/table.md

```diff
@@ -439,6 +742,155 @@ export default {
             themeColors: ['green', 'yellow']
           }
         ]
+      },
```

Comment:
We need to show a demo of clickable links inside a row thats clickable (If I have an external link inside a clickable row it shouldn't open the row too). 

We also need an example of a button with an action (kpop, alert window, etc) to demo you can click a button without opening the row

Reply:
We have a demo here.

![image](https://user-images.githubusercontent.com/2403177/86624577-6635bd00-bf78-11ea-9dc0-6e48528abcdc.png)

I'll add an example with button too

Reply:
Example with buttons added. Note that everything uses the same row click handler; anything more fine tuned can also be added in the row data's metadata (like if the button goes to a different href) but its all implementation dependent. 

![image](https://user-images.githubusercontent.com/2403177/86669937-f5f35f80-bfa8-11ea-9241-90e5bdc2b343.png)

Reply:
This doesn't show that clicking the button doesn't fire the row click event. The button should instead fire an alert, console.log, Kpop, etc. Something other than a row click.

Overall I think these docs are getting more complicated then needed. Lets pair tomorrow to make sure we're all on the same page

Reply:
How many examples do these docs need to show? I mean we can show a link, a button etc doing something entirely different too, but like I said this is all implementation specific. For this example, it fires the row click. If we want the button or link to do something else, we can just add an `@click` handler to them and then in the row event handler, add something like `if (e.target.tagName !== 'BUTTON')` to prevent the row click for buttons etc but again this is all up to how complicated and how many examples we need in the doc

Reply:
Only one example is needed. The idea is to show that  while a row has a `row:click` event, having another action (like an actions menu) does not fire the row event. An example of this would be having the row click event open a new window and the button in a cell fires an alert, *or* KPop showing the new window does not open (row event doesn't fire). We do not want to add an  `e.target !==` in implementation. It is currently developed so that is not needed.

Reply:
`It is currently developed so that is not needed.` I don't understand what is meant by this. Currently the way it is, having any link or button inside a row with a row handler will also trigger the row handler, so we need to handle that case to prevent the row handler from firing the row event. That's the whole point of the example right? But there's nothing in KTable that would inherently prevent this, so we need to deal with this in the implementation.

Reply:
this code [here](https://github.com/Kong/kongponents/pull/237/files#diff-158ff5639cfcb0ec94ce4226ff60c18dR204) is supposed to not fire the TD click. So a link or a button _inside_ of a row should not fire the event. @darrenjennings helped write this code before this ticket was started

Reply:
I don't think that should be inside `KTable` it should be implementation specific because we don't know what tags or listeners the end user will put in. `TD` and `mouseup` are assumptions based on something that should be implementation specific. In the example we would have to add an additional check for `A` or `BUTTON` since `TD` is not the target element in those cases, and an additional `click` listener to pluck. We can discuss with @darrenjennings but my feeling is once it gets this complex and specific then it should be left to the implementation level.

Reply:
The click target is always TD unless it’s a link or a button. So that’s why it’s implemented in the table. If you click a link or a button it should not trigger the row event. The only implementation addition should be meta key if the user wants. This already was discussed and implemented

Reply:
I see, I didn't have the entire background on the implementation of `trlisteners` but it seems like it does work the way you described once I changed `mouseup` to `click` because in the examples we are using click listeners, not mouseup. I'll revise the examples.